### PR TITLE
An optimization about MessageFuture

### PR DESCRIPTION
**Pull request by XCXCXCXCX**
### Ⅰ. Describe what this PR did
In the MessageFuture class, it is currently implemented using CountDownLatch.
This synchronization method is relatively heavy, we should replace it with CompletableFuture to achieve.
1. Will not change the MessageFuture API
2. Consider that the server will process a large number of Messages in the future, and the Get mode of the CompletableFuture will perform better when testing a large number of messages.

### CODE REVIEW

File: core/src/main/java/io/seata/core/protocol/MessageFuture.java [hunk 1]
```diff
@@ -26,13 +29,12 @@
  * @author jimin.jm @alibaba-inc.com
  * @date 2018 /10/9
  */
-public class MessageFuture {
+public class MessageFuture{
```
Comment (CoffeeLatte007) [inline core/src/main/java/io/seata/core/protocol/MessageFuture.java]: { white space

Comment (github-ygy) [inline core/src/main/java/io/seata/core/protocol/MessageFuture.java]: add a space right the class

File: core/src/main/java/io/seata/core/protocol/MessageFuture.java [hunk 2]
```diff
@@ -54,18 +56,22 @@ public boolean isTimeout() {
...
     public Object get(long timeout, TimeUnit unit) throws TimeoutException,
         InterruptedException {
-        boolean success = latch.await(timeout, unit);
-        if (!success) {
+        Object result = null;
+        try {
+            result = origin.get(timeout, unit);
+        } catch (ExecutionException e) {
+            throw new ShouldNeverHappenException("Should not get results in a multi-threaded environment", e);
+        } catch (TimeoutException e){
```
Comment (CoffeeLatte007) [inline core/src/main/java/io/seata/core/protocol/MessageFuture.java]: { white space

File: core/src/main/java/io/seata/core/protocol/MessageFuture.java (hunk 1 above)
Comment (XCXCXCXCX) [reply core/src/main/java/io/seata/core/protocol/MessageFuture.java]: OK

File: core/src/main/java/io/seata/core/protocol/MessageFuture.java (hunk 2 above)
Comment (XCXCXCXCX) [reply core/src/main/java/io/seata/core/protocol/MessageFuture.java]: OK

File: core/src/main/java/io/seata/core/protocol/MessageFuture.java (hunk 1 above)
Comment (XCXCXCXCX) [reply core/src/main/java/io/seata/core/protocol/MessageFuture.java]: OK

File: core/src/main/java/io/seata/core/protocol/MessageFuture.java [hunk 3]
```diff
@@ -74,8 +80,7 @@ public Object get(long timeout, TimeUnit unit) throws TimeoutException,
      * @param obj the obj
      */
     public void setResultMessage(Object obj) {
-        this.resultMessage = (obj == null ? NULL : obj);
-        latch.countDown();
+        origin.complete(obj == null ? NULL : obj);
```
Comment (github-ygy) [inline core/src/main/java/io/seata/core/protocol/MessageFuture.java]: how about remove the Null ？

**Comment by github-ygy** [comment]
@CoffeeLatte007 do u  have any other questions ？

**Pull request merged**